app/utils/configuration/decorators.py

```diff
@@ -0,0 +1,16 @@
+from functools import wraps
+
+
+def authorised(func):
+    def authoriser(name=None):
+        @wraps(func)
+        async def decorated_function(request, *args, **kwargs):
+            # if 'upload' in request.url: TO DO: mapping to permission/role
+            # system
+            response = f(request, *args, **kwargs)
```

Comment:
@thomasmastersmethods: What is `f` in this context? Is it bound to anything? And where does this throw unauthenticated users out?